### PR TITLE
refactor(annotation): Uses JDT visitor to build Java 8 annotations.

### PR DESCRIPTION
**Pull request by GerardPaligot**

### CODE REVIEW

File: src/main/java/spoon/reflect/reference/CtArrayTypeReference.java [hunk 1]
```diff
@@ -27,6 +27,11 @@
 	CtTypeReference<?> getComponentType();
 
 	/**
+	 * Gets the type reference of the array.
+	 */
+	CtTypeReference<?> getComponentTypeReference();
```
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/reference/CtArrayTypeReference.java]: Change to `getArrayType` + example in javadocs.

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

File: src/main/java/spoon/reflect/reference/CtArrayTypeReference.java [hunk 2]
```diff
@@ -23,10 +23,19 @@
...
 	 * Gets the type of the elements contained in this array.
+	 * e.g., if you have the array <code>int[][][]</code>,
+	 * this method returns a type reference for <code>int[][]</code>.
 	 */
 	CtTypeReference<?> getComponentType();
 
 	/**
+	 * Gets the type reference of the array.
+	 * e.g., if you have the array <code>int[][][]</code>,
+	 * this method returns an integer in the type reference.
```
Comment (monperrus) [inline src/main/java/spoon/reflect/reference/CtArrayTypeReference.java]: this method returns a type reference to "int".

File: src/main/java/spoon/reflect/reference/CtArrayTypeReference.java [hunk 3]
```diff
@@ -23,10 +23,19 @@
 
 	/**
 	 * Gets the type of the elements contained in this array.
+	 * e.g., if you have the array <code>int[][][]</code>,
+	 * this method returns a type reference for <code>int[][]</code>.
 	 */
 	CtTypeReference<?> getComponentType();
 
 	/**
+	 * Gets the type reference of the array.
```
Comment (monperrus) [inline src/main/java/spoon/reflect/reference/CtArrayTypeReference.java]: Gets the type of the array elements at the finest grain

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Pull request merged**